src/source/flush.rs

```diff
@@ -1,43 +1,49 @@
 use metric;
 use mio;
-use source::Source;
+use source;
 use std::thread::sleep;
 use std::time::Duration;
+use thread;
 use util;
 use util::send;
 
-/// The source of all flush pulses. See `FlushTimer::run` for more details.
+/// The source of all flush pulses.
 pub struct FlushTimer {
     chans: util::Channel,
 }
 
-impl FlushTimer {
+/// Nil config for FlushTimer.
+pub struct FlushTimerConfig;
+
+fn tick_tock(mut chans: util::Channel, _poll: mio::Poll) {
```

Comment:
Okay, I guess I see what's going on. Rather than have a single function that we pass into a thread – jettisoning the impl in the process – I'd rather see a function on the impl that gets passed into a thread.

Reply:
Which impl?  Source?

Reply:
Yeah, probably Source. The appeal, to me, of encoding all this into traits – rather than mostly similar functions – is that it should make cranking out a new source / sink / filter straightforward. Plus, the compiler checks up on you.

Reply:
Lets consolidate this discussion in the thread in internal.rs.